### PR TITLE
Refactor user controller tests

### LINKED ISSUE #443: Refactor User Controller Tests with helpers
# Problem

Adding helpers from api_case.ex to the tests for User Controller.

## References

Progress on: #413

### PR DESCRIPTION
**Pull request by mackenziehicks**
# What's in this PR?

Added helpers from api_case.ex to the tests for User Controller.

## References
Fixes #443 

Progress on: #413

### CODE REVIEW

File: test/controllers/user_controller_test.exs [hunk 1]
```diff
@@ -135,19 +127,7 @@ defmodule CodeCorps.UserControllerTest do
...
-
-      id = json_response(conn, 200)["data"]["id"]
-      assert id
-      user =  Repo.get(User, id)
-      assert user.email == "test@user.com"
-      assert user.first_name == "Test"
-      assert user.last_name == "User"
-      assert user.website == "http://www.example.com"
-      assert user.biography == "Just a test user"
+    assert conn |> authenticate(user) |> put(path, params) |> json_response(200)
```
Comment (begedin) [inline test/controllers/user_controller_test.exs]: Got a bad indent here

**Review by begedin — CHANGES_REQUESTED** [review]

**Comment by begedin** [comment]
Tests seem to pass here. What's the reason for "help wanted". Anything I can help with?

File: test/controllers/user_controller_test.exs [hunk 2]
```diff
@@ -27,68 +27,61 @@ defmodule CodeCorps.UserControllerTest do
   describe "index" do
 
     test "lists all entries on index", %{conn: conn} do
-      conn = get conn, user_path(conn, :index)
-      assert json_response(conn, 200)["data"] == []
+      [user_1, user_2] = insert_pair(:user)
+
+      conn
+        |> request_index
```
Comment (sbatson5) [inline test/controllers/user_controller_test.exs]: I think the patter is not to indent here.

File: test/controllers/user_controller_test.exs [hunk 3]
```diff
@@ -27,68 +27,61 @@ defmodule CodeCorps.UserControllerTest do
...
-      json = conn |> authenticate(user) |> get(path) |> json_response(200)
+    @tag :authenticated
+    test "renders email when authenticated", %{conn: conn, current_user: current_user} do
+      json =
+        conn
+        |> request_show(current_user)
+        |> json_response(200)
 
-      assert json["data"]["attributes"]["email"] == user.email
+      assert json["data"]["attributes"]["email"] == current_user.email
```
Comment (sbatson5) [inline test/controllers/user_controller_test.exs]: Do we need to assert that the email is serialized correctly?  Does the json_response returning 200 cover us?

**Review by sbatson5 — COMMENTED** [review]
This is looking good to me 👍

File: test/controllers/user_controller_test.exs (hunk 3 above)
Comment (joshsmith) [reply test/controllers/user_controller_test.exs]: @sbatson5 I initially thought that we did need to test this here, but we do have a way of testing this at the level of the view test. In that test we do `conn = Phoenix.ConnTest.build_conn |> assign(:current_user, auth_user)` in order to check that the user only see their email when authorized.

We can remove that assertion from here then @mackenziehicks. Asserting `200` is sufficient.

**Review by begedin — APPROVED** [review]
👍

**Pull request merged**